cdisc_rules_engine/utilities/data_processor.py

```diff
@@ -170,6 +183,38 @@ def variable_names(self, params: OperationParams) -> set:
                 ).keys()
             )
 
+    def get_column_order_from_library(self, params: OperationParams) -> pd.Series:
+        """
+        Fetches column order for a given domain from the CDISC library.
+        Returns it as a Series of lists like:
+        0    ["STUDYID", "DOMAIN", ...]
+        1    ["STUDYID", "DOMAIN", ...]
+        2    ["STUDYID", "DOMAIN", ...]
+        ...
+
+        Length of Series is equal to the length of given dataframe.
+        The lists with column names are sorted
+        in accordance to "ordinal" key of library metadata.
+        """
+        # get dataset class
+        dataset_class: str = self.data_service.get_dataset_class(
+            params.dataframe, params.dataset_path, params.datasets
+        )
+
+        # get variables metadata from the standard model
+        variables_metadata: List[
+            dict
+        ] = self._get_variables_metadata_from_standard_model(
+            dataset_class, params.domain, params.standard, params.standard_version
+        )
+
+        # create a list of variable names in accordance to the "ordinal" key
+        variables_metadata.sort(key=lambda item: item["ordinal"])
+        variable_names_list = [
```

Comment:
There is one more step to getting the correct order. If the class is one of Interventions, Findings, or Events then the  General Observation class variables need to be added to the variable names list. This should be done by adding the variables with role "Identifier" to the beginning of the list and variables with role "Timing" to the end of the list. If we are not caching the role, we should be.

Reply:
Also, this operation probably should actually do the comparison, because some variables can be omitted. So it should  check whether or not the dataframe columns are sorted based on the supplied variables ordinals. Not sure if you are doing this elsewhere though

Reply:
Okay, I understand, I will modify the logic. 

The actual comparison is done in "conditions", the operations just return the variable list in a given order and the conditions compare the columns. You can check the example rule in conftest.py. Is this approach okay for you? I want the operation to just return a list and not compare anything so that we could reuse its result for further validation scenarios if needed.

Reply:
That approach is ok with me if it can handle missing variables that are allowed to be missing. For example if the returned order is ["STUDYID", "DOMAIN", "USUBJID", "VAR1", "VAR2", "VAR3", "VAR4"] but VAR3 is permissible then this is still a valid variable order for the dataset to have ["STUDYID", "DOMAIN", "USUBJID", "VAR1", "VAR2", "VAR4"]. To be clear I don't think this operation/check needs to know what variables are permissible or not, but it should be able to tell if any set of variables is in order based on their ordinals.

Reply:
Updated the logic

Reply:
I don't see how you would write an rule that allows for permissible variables to be missing. Can you provide an example?

Reply:
There is no such option right now, I would suggest implementing it in a separate PR not to complicate this one

Reply:
That works for me. Approved